src/main/java/com/xpanxion/java/assignments/student1/Worker1.java

```diff
@@ -33,4 +37,23 @@ public void ex3() {
         var result = products.stream().filter(p -> p.getPrice() >= 10.0);
         System.out.println(result.collect(Collectors.toList()));
     }
+
+    public void ex4() {
+        var products = DataAccess.getProducts();
+        Optional<Float> total;
+        NumberFormat fomatter = NumberFormat.getCurrencyInstance();
+        String moneyString;
+
+        Map<Integer, Department> departmentMap = DataAccess.getDepartments().stream()
+                .collect(Collectors.toMap(Department::getId, Function.identity()));
+
+        total = products.stream()
+                .peek(product -> product.setDepartmentName(departmentMap.get(product.getDepartmentId()).getName()))
```

Comment:
Let's avoid the use of `.peek()` because it's there for debugging purposes only.  I know the IDE recommends it, but it's not good to use peek in production.